core/trino-spi/src/main/java/io/trino/spi/type/Type.java

```diff
@@ -97,6 +97,14 @@ default TypeOperatorDeclaration getTypeOperatorDeclaration(TypeOperators typeOpe
      */
     BlockBuilder createBlockBuilder(BlockBuilderStatus blockBuilderStatus, int expectedEntries);
 
+    /**
+     * Returns the nearest greater value (if exists).
+     */
+    default Optional<?> getNearestGreaterValue(Object value)
+    {
+        return Optional.empty();
```

Comment:
Good question - currently, I don't think we need to distinguish between the cases.
Hipothetically, this could be helpful to support expanding `[125, inf)` range into `{125, 126, 127}` discrete set (which is not supported in this PR).

To distinguish between the cases, we can add a new method (e.g. `supportsNearestGreaterValue`) returning a boolean, or throw an `UnsupportedOperationException` for unsupported types. Not sure which is better TBH...